pages/[[...slug]].js

```diff
@@ -31,26 +41,48 @@ export const getStaticPaths = async () => {
 };
 
 export const getStaticProps = async (context) => {
-    const {
-        params: { slug }
-    } = context;
     try {
-        let url = slug ? `/${slug.join('/')}` : '/index';
+        const {
+            params: { slug }
+        } = context;
+
+        // Determine our head and tail
+        const [head, ...tail] = slug || [];
+        const languageProps = await getLanguagesProps(head);
+        let category;
+        // Build our URL
+        // if the hasLanguages predicate returns true then join with the tail otherwise join the slugs array
+        let url = slug
+            ? languageProps.hasLanguages
+                ? `/${tail.join('/')}`
+                : `/${slug.join('/')}`
+            : '/index';
+
+        if (slug?.includes('category')) {
```

Comment:
Need to add a comment here to explain why we're doing this, basically for our tag url system.